### PR TITLE
UCP/STATIC: added ucp test

### PR DESCRIPTION
**Pull request by hoopoepg**
- added ucp test for static build
- added CMA transport into static test
- updated docker config to enable CMA

### CODE REVIEW

File: buildlib/pr/main.yml [hunk 1]
```diff
@@ -194,8 +196,9 @@ stages:
             centos7:
               CONTAINER: centos7_ib
               test_static: yes
-              extra_modules: ucx-ib
-              extra_tls: dc_mlx5 rc_mlx5 ud_mlx5 rc_verbs ud_verbs
+              extra_modules: ucx-ib ucx-cma
+              extra_tls: dc_mlx5 rc_mlx5 ud_mlx5 rc_verbs ud_verbs cma
+              run_tls: ib rc rc_v rc_x dc dc_x ud ud_v ud_x shm sm
```
Comment (yosefe) [inline buildlib/pr/main.yml:201]: 👍

File: buildlib/tools/builds.sh [hunk 2]
```diff
@@ -407,6 +407,14 @@ build_static() {
 
 	${WORKSPACE}/buildlib/tools/check_tls.sh $EXTRA_TLS
 
+	for tls in tcp $RUN_TLS; do
+		echo UCX_TLS=$tls
+		UCX_TLS=$tls ./ucp_hello_world_static &
```
Comment (yosefe) [inline buildlib/tools/builds.sh:412]: 1. can we derive the TCP port number from agent id, like in test_jenkins.sh
since we share net namespace with host, it can create collisions

2. we should move the static test from builds.sh to a separate script since it doesn't just "build" now, it also runs tests. can be in this or another PR.

Comment (hoopoepg) [reply buildlib/tools/builds.sh:412]: > * can we derive the TCP port number from agent id, like in test_jenkins.sh
>   since we share net namespace with host, it can create collisions

should not be issue because containers are launched in separate network namespaces and run in single thread

> * we should move the static test from builds.sh to a separate script since it doesn't just "build" now, it also runs tests. can be in this or another PR.

will do later - need deeper changes (different infrastructure for variables and container initialization)

Comment (yosefe) [reply buildlib/tools/builds.sh:412]: > should not be issue because containers are launched in separate network namespaces and run in single thread

Actually, we run them with "-net host" in Azure

Comment (hoopoepg) [reply buildlib/tools/builds.sh:412]: > Actually, we run them with "-net host" in Azure

as I can see it is launched in own network namespace:
```/scrap/bin/docker create --label 8f00dc --network vsts_network_e782202237e342969d72245c7a75d492 -v /hpc/local:/hpc/local ... ```

@avildema any ideas?

Comment (yosefe) [reply buildlib/tools/builds.sh:412]: `/scrap/bin/docker` is wrapper to use host net

Comment (hoopoepg) [reply buildlib/tools/builds.sh]: hmmm, does it just removes --network argument?
but ok, added custom server port

Comment (avildema) [reply buildlib/tools/builds.sh]: you should add to command like --het host 
like this https://github.com/openucx/ucx/blob/master/buildlib/pr/main.yml#L16

Comment (hoopoepg) [reply buildlib/tools/builds.sh]: > you should add to command line `--net=host `

we need own network namespace for docker containers

Comment (yosefe) [reply buildlib/tools/builds.sh]: > we need own network namespace for docker containers

in general yes but it had some issues with TCP tests using bridge netorks (though i think it should be resolved now)
in any case, to be on safe side, IMO better generate TCP port num from agent id in this PR

Comment (hoopoepg) [reply buildlib/tools/builds.sh]: yep, done.
under testing...

**Review by yosefe — APPROVED** [review]

**Review by brminich — APPROVED** [review]

**Review by yosefe — DISMISSED** [review]

**Review by yosefe — APPROVED** [review]

**Comment by yosefe** [comment]
@hoopoepg can you pls squash?

File: test/apps/uct_info/Makefile.in [hunk 3]
```diff
@@ -4,19 +4,24 @@
 # See file LICENSE for terms.
 #
 
-all: uct_info uct_info_static
+all: uct_info uct_info_static ucp_hello_world_static
```
Comment (brminich) [inline test/apps/uct_info/Makefile.in:7]: would not it be better to define static target in `ucp_hello_world` makefile?
Looks weird that `uct_info` app also builds `ucp_hello_world`

Comment (yosefe) [reply test/apps/uct_info/Makefile.in:7]: yes, since we need to merge this one to fix test issues

**Review by yosefe — APPROVED** [review]
